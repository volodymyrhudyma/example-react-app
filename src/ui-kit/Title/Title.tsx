import React, { ReactNode } from 'react';
import { Wrapper } from './styled/Title.styled';

interface ITitleProps {
  type: string;
  children: ReactNode;
}

const Title = ({ type, children }: ITitleProps) => (
  <Wrapper component={type}>{children}</Wrapper>
);

Title.defaultProps = {
  type: 'h1',
};

export default Title;
