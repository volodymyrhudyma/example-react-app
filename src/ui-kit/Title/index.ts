import Component from './Title';

export default Component;
