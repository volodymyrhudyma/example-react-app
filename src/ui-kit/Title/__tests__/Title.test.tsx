import React from 'react';
import { shallow } from 'enzyme';
import Title from '../Title';

describe('Title component', () => {
  it('should render correctly', () => {
    const component = shallow(<Title>Card</Title>);
    expect(component).toMatchSnapshot();
  });
});
