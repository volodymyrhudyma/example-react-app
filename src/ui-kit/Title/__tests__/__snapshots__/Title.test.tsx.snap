// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render correctly 1`] = `
<Styled(Component)
  component="h1"
>
  Card
</Styled(Component)>
`;
